openmdao/core/system.py

```diff
@@ -5367,7 +5367,7 @@ def set_val(self, name, val, units=None, indices=None):
         n_proms = 0  # if nonzero, name given was promoted input name w/o a matching prom output
 
         try:
-            ginputs = model._group_inputs
+            ginputs = self._group_inputs
```

Comment:
should probably update/remove the comment in the `except` block....  it implies that the previous behavior was intended (i.e. only looking at the top level)

Reply:
done